util/settings/settings_test.go

```diff
@@ -1858,6 +1858,49 @@ func TestIsImpersonationEnabled(t *testing.T) {
 		"when user enables the flag in argocd-cm config map, IsImpersonationEnabled() must not return any error")
 }
 
+func TestExternalRevisionConsideredOverride(t *testing.T) {
+	// When there is no argocd-cm itself,
+	// Then ExternalRevisionConsideredOverride() must return false (default value) and an error with appropriate error message.
+	kubeClient := fake.NewClientset()
+	settingsManager := NewSettingsManager(t.Context(), kubeClient, "default")
+	featureFlag, err := settingsManager.ExternalRevisionConsideredOverride()
+	require.False(t, featureFlag,
+		"with no argocd-cm config map, ExternalRevisionConsideredOverride() must return return false (default value)")
+	require.ErrorContains(t, err, "configmap \"argocd-cm\" not found",
+		"with no argocd-cm config map, ExternalRevisionConsideredOverride() must return an error")
+
+	// When there is no impersonation feature flag present in the argocd-cm,
+	// Then ExternalRevisionConsideredOverride() must return false (default value) and nil error.
+	_, settingsManager = fixtures(map[string]string{})
+	featureFlag, err = settingsManager.ExternalRevisionConsideredOverride()
+	require.False(t, featureFlag,
+		"with empty argocd-cm config map, ExternalRevisionConsideredOverride() must return false (default value)")
+	require.NoError(t, err,
+		"with empty argocd-cm config map, ExternalRevisionConsideredOverride() must not return any error")
+
+	// When user disables the feature explicitly,
+	// Then ExternalRevisionConsideredOverride() must return false and nil error.
+	_, settingsManager = fixtures(map[string]string{
+		"application.sync.externalRevisionConsideredOverride": "false",
+	})
+	featureFlag, err = settingsManager.ExternalRevisionConsideredOverride()
+	require.False(t, featureFlag,
+		"when user enables the flag in argocd-cm config map, ExternalRevisionConsideredOverride() must return user set value")
+	require.NoError(t, err,
+		"when user enables the flag in argocd-cm config map, ExternalRevisionConsideredOverride() must not return any error")
+
+	// When user enables the feature explicitly,
+	// Then ExternalRevisionConsideredOverride() must return true and nil error.
+	_, settingsManager = fixtures(map[string]string{
+		"application.sync.externalRevisionConsideredOverride": "true",
+	})
+	featureFlag, err = settingsManager.ExternalRevisionConsideredOverride()
+	require.True(t, featureFlag,
+		"when user enables the flag in argocd-cm config map, ExternalRevisionConsideredOverride() must return user set value")
+	require.NoError(t, err,
```

Comment:
can we please turn this test to multiple test cases with proper names explaining what we test each time and remove the comments?

This will make the code self-descriptive and will follow golang standard guidelines that recommend to split test cases in the same function